### PR TITLE
DFPL-2072-SendAndReplyMessage

### PR DESCRIPTION
**Pull request by prabhamuthu15**
### JIRA link (if applicable) ###
[DFPL-2072](https://tools.hmcts.net/jira/browse/DFPL-2072)

### Change description ###

Write UI test for send and reply message

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by AlistairEdwinOsborne — DISMISSED** [review]
Looks good to me! but if you've got some time spare would it be possible to get a demo?

File: Jenkinsfile_nightly [hunk 1]
```diff
@@ -30,7 +30,9 @@ def integrationTestSecrets = [
   'fpl-aat': [
     secret('integration-test-notify-service-key', 'INTEGRATION_TEST_NOTIFY_SERVICE_KEY'),
     secret('docmosis-api-key', 'INTEGRATION_TEST_DOCMOSIS_TORNADO_KEY'),
-    secret('e2e-test-password', 'E2E_TEST_PASSWORD')
+    secret('e2e-test-password', 'E2E_TEST_PASSWORD'),
+    secret('system-update-user-username', 'SYSTEM_UPDATE_USER_USERNAME'),
```
Comment (Iphiclus) [inline Jenkinsfile_nightly:34]: @prabhamuthu15 do these need to be added to README.md for .env info for people?

File: playwright-e2e/pages/base-page.ts
Comment (Iphiclus) [inline playwright-e2e/pages/base-page.ts:1]: @prabhamuthu15 minor style/format fixes?

File: playwright-e2e/pages/judicialMessage.ts
Comment (Iphiclus) [inline playwright-e2e/pages/judicialMessage.ts:1]: @prabhamuthu15 minor style/format fixes?

Also name of file do we want to keep camel case or kebab case/spinal case ?

File: playwright-e2e/settings/userCredentials.ts
Comment (Iphiclus) [inline playwright-e2e/settings/userCredentials.ts:1]: @prabhamuthu15 minor style/format fixes?

Also name of file do we want to keep camel case or kebab case/spinal case ?

File: playwright-e2e/tests/messageJudge.spec.ts
Comment (Iphiclus) [inline playwright-e2e/tests/messageJudge.spec.ts:1]: @prabhamuthu15 minor style/format fixes?

Also name of file do we want to keep camel case or kebab case/spinal case ?

File: playwright-e2e/tests/messageJudge.spec.ts [hunk 2]
```diff
@@ -1,30 +1,67 @@
...
+      casename = 'CTSC Admin Close Message' + dateTime.slice(0, 10);
+      await apiDataSetup.updateCase(casename,caseNumber,caseDataCloseMessage);
+      await  signInPage.visit();
+      await signInPage.login(CTSCUser.email,CTSCUser.password);
+      await signInPage.navigateTOCaseDetails(caseNumber);
+      await judicialMessages.gotoNextStep('Reply to messages');
+      await judicialMessages.CTSCUserCloseMessage();
+      await judicialMessages.checkYourAnsAndSubmit();
+      await judicialMessages.tabNavigation('Judicial messages');
+    //  await expect(page.getByText('Closed')).await page.locator('ccd-read-complex-field-table ccd-read-collection-field ccd-field-read-label div').click();
```
Comment (Iphiclus) [inline playwright-e2e/tests/messageJudge.spec.ts:55]: @prabhamuthu15 do we need this commented out code?

**Review by Iphiclus — CHANGES_REQUESTED** [review]

File: Jenkinsfile_nightly (hunk 1 above)
Comment (prabhamuthu15) [reply Jenkinsfile_nightly:34]: Updated in Readme

File: playwright-e2e/settings/userCredentials.ts
Comment (prabhamuthu15) [reply playwright-e2e/settings/userCredentials.ts:1]: The file is same as master, do you want to update it ?

File: playwright-e2e/pages/base-page.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+async clickContinue(){
+  await this.continue.click();
+
+}
+
+
+}
+
+
```
Comment (DanCatchpole) [inline playwright-e2e/pages/base-page.ts:42]: Think the indentation in this file needs to be updated

File: playwright-e2e/tests/judicial-message.spec.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    })
+
+});
+
+
+
+
+
+
+
```
Comment (DanCatchpole) [inline playwright-e2e/tests/judicial-message.spec.ts:69]: Couple empty lines to remove here

File: playwright-e2e/tests/judicial-message.spec.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        await  signInPage.visit();
+        await signInPage.login(judgeUser.email,judgeUser.password);
+        await signInPage.navigateTOCaseDetails(caseNumber);
+        await judicialMessages.gotoNextStep('Reply to messages');
+        await judicialMessages.judgeReplyMessage();
+        await judicialMessages.checkYourAnsAndSubmit();
+        await judicialMessages.tabNavigation('Judicial messages');
+        await expect(page.getByText('FamilyPublicLaw+ctsc@gmail.com - Some note judiciary-only@mailnesia.com - Reply CTSC admin about the hearing.')).toBeVisible();
+    })
+    test('CTSC admin close the Message',async({page,signInPage,judicialMessages}) =>{
```
Comment (DanCatchpole) [inline playwright-e2e/tests/judicial-message.spec.ts:47]: also a newline break separating the tests would be good here for readability!

File: playwright-e2e/settings/user-credentials.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+
+
+const e2ePw = process.env.E2E_TEST_PASSWORD || 'Password1234';
+const defaultPwd = process.env.SYSTEM_UPDATE_USER_PASSWORD || 'Password12';
```
Comment (DanCatchpole) [inline playwright-e2e/settings/user-credentials.ts:4]: think these might need to be reset back to `|| ''`;, for fortify?

File: Jenkinsfile_nightly [hunk 7]
```diff
@@ -78,19 +80,8 @@ withNightlyPipeline(type, product, component) {
   env.ORG_ID = params.ORG_ID
 
   loadVaultSecrets(integrationTestSecrets)
-
-  enableMutationTest()
-  enableFortifyScan('fpl-aat')
```
Comment (DanCatchpole) [inline Jenkinsfile_nightly:83]: these have been removed rather than uncommented out!

**Review by DanCatchpole — APPROVED** [review]

**Pull request merged**